### PR TITLE
simplify default config path

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We were complicating the default config parameter in `v5`. First of all, we shouldn't ask for a directory and should ask for a file path to store config file. I'm not going to insist on that but still 1/5. 

The other thing was, to not break compatibility in `v5`, we were looking for `$HOME/.mmctl` file before everything else and use that file if there is one. Now it's a good time to change this with breaking the compatibility and let it be explicit. We can now use the  default `$XDG_CONFIG_HOME` to store config files. 

And last but not least let's be explicit about the flag name, we should use `--config-file` or `--config-file-path`, I'm 0/5 on either. Want to hear other opinions as well.

https://mattermost.atlassian.net/browse/MM-38036

#### Release Note

```release-note
mmctl will start using $XDG_CONFIG_HOME/mmctl/config file as a config file parameter, please consider updating the flag and value if you want to use your existing config file.
```

### CODE REVIEW

**Comment by streamer45** [comment]
I am wondering if the original plan was to store more than one file? That could be a reason to use a directory instead of a single file. Anyhow, I'd argue that we could even make it simpler and go for `--config` since it's accompanied by documentation explaining in detail what it refers to. I believe conciseness to be an extremely important value when on the command line.

**Comment by isacikgoz** [comment]
Agreed, we can use `--config` as well. In fact, it is more commonly used like that. For using multiple files, I believe we can add more along the way if they are ever needed. So far single file did work 😅

**Comment by isacikgoz** [comment]
Thought about it, would it confuse users to have a subcommand as `config` and a flag as `--config` at the same time? We have a config subcommand for MM config related stuff.

**Comment by streamer45** [comment]
> Thought about it, would it confuse users to have a subcommand as `config` and a flag as `--config` at the same time? We have a config subcommand for MM config related stuff.

I wouldn't think so. The flag is global and clearly refers to the tool itself.

File: commands/auth_utils.go [hunk 1]
```diff
@@ -47,33 +54,52 @@ func init() {
...
 	return configPath
 }
 
 func resolveConfigFilePath() string {
-	configPath := getDefaultConfigPath()
-	if p := viper.GetString("config-path"); p != xdgConfigHomeVar {
-		configPath = strings.Replace(viper.GetString("config-path"), userHomeVar, currentUser.HomeDir, 1)
-	}
+	// we warn users that config-path is deprecated
+	suppressWarnings := viper.GetBool("suppress-warnings")
```
Comment (streamer45) [inline commands/auth_utils.go:83]: I am wondering if we could follow other commands and name this `--quiet`.

**Review by streamer45 — APPROVED** [review]
Nicely done :+1:

**Review by jwilander — APPROVED** [review]

File: commands/auth_utils.go (hunk 1 above)
Comment (isacikgoz) [reply commands/auth_utils.go:83]: `--quiet` may have some other context, it is mostly used for silencing all of the output but this is for rather suppressing warning messages.

Comment (streamer45) [reply commands/auth_utils.go:83]: That would be `--silent` :p Just kidding, this is nothing blocking, just a thought.

Comment (isacikgoz) [reply commands/auth_utils.go:83]: I get it covered :p https://github.com/mattermost/mmctl/pull/371

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - I'll test this after merge 👍

**Pull request merged**